Our Nightly Quarkus build deploy to Nexus was failing after the gradle upgrade to 8.9

### Describe the bug

From the last gradle bump upgrade to 8.9 (https://github.com/quarkusio/quarkus/pull/41953) our nighly jenkins jobs is failing on quarkus-project-core-extension-codestarts and it worked with gradle 8.8 version.

Logs with the last gradle upgrade(8.9) :

```
22:42:40 [INFO] 
22:42:40 [INFO] --------< io.quarkus:quarkus-project-core-extension-codestarts >--------
22:42:40 [INFO] Building Quarkus - Dev tools - Project Core Extension Codestarts 999-SNAPSHOT [909/1293]
22:42:40 [INFO] --------------------------------[ jar ]---------------------------------
22:42:40 [INFO] 
22:42:40 [INFO] --- buildnumber-maven-plugin:3.0.0:create (get-scm-revision) @ quarkus-project-core-extension-codestarts ---
22:42:40 [INFO] 
22:42:40 [INFO] --- formatter-maven-plugin:2.24.1:format (default) @ quarkus-project-core-extension-codestarts ---
22:42:40 [INFO] Loaded from the build cache
22:42:40 [INFO] 
22:42:40 [INFO] --- impsort-maven-plugin:1.11.0:sort (sort-imports) @ quarkus-project-core-extension-codestarts ---
22:42:40 [INFO] Processed 0 files in 00:00.000 (Already Sorted: 0, Needed Sorting: 0)
22:42:40 [INFO] 
22:42:40 [INFO] --- maven-resources-plugin:3.3.1:resources (default-resources) @ quarkus-project-core-extension-codestarts ---
22:42:40 [INFO] Copying 182 resources from src/main/resources to target/classes
22:42:40 [INFO] Copying 1 resource from src/main/filtered to target/classes
22:42:40 [INFO] The encoding used to copy filtered properties files have not been set. This means that the same encoding will be used to copy filtered properties files as when copying other filtered resources. This might not be what you want! Run your build with --debug to see which files might be affected. Read more at https://maven.apache.org/plugins/maven-resources-plugin/examples/filtering-properties-files.html
22:42:40 [INFO] 
22:42:40 [INFO] --- maven-compiler-plugin:3.13.0:compile (default-compile) @ quarkus-project-core-extension-codestarts ---
22:42:40 [INFO] No sources to compile
22:42:40 [INFO] 
22:42:40 [INFO] --- maven-resources-plugin:3.3.1:testResources (default-testResources) @ quarkus-project-core-extension-codestarts ---
22:42:40 [INFO] skip non existing resourceDirectory /home/hudson/hudson_workspace/workspace/utils-deploy-quarkus-main-artifacts-to-internal-repository/devtools/project-core-extension-codestarts/src/test/resources
22:42:40 [INFO] 
22:42:40 [INFO] --- maven-compiler-plugin:3.13.0:testCompile (default-testCompile) @ quarkus-project-core-extension-codestarts ---
22:42:40 [INFO] No sources to compile
22:42:40 [INFO] 
22:42:40 [INFO] --- maven-surefire-plugin:3.2.5:test (default-test) @ quarkus-project-core-extension-codestarts ---
22:42:40 [INFO] No tests to run.
22:42:40 [INFO] Skipped
22:42:40 [INFO] 
22:42:40 [INFO] --- exec-maven-plugin:3.1.0:exec (generate-gradle-wrapper) @ quarkus-project-core-extension-codestarts ---
22:42:41 To honour the JVM settings for this build a single-use Daemon process will be forked. For more on this, please refer to https://docs.gradle.org/8.9/userguide/gradle_daemon.html#sec:disabling_the_daemon in the Gradle documentation.
22:42:42 Daemon will be stopped at the end of the build 
22:42:51 > Task :init FAILED
22:42:51 
22:42:51 FAILURE: Build failed with an exception.
22:42:51 
22:42:51 * What went wrong:
22:42:51 Execution failed for task ':init'.
22:42:51 > Aborting build initialization due to existing files in the project directory: '/home/hudson/hudson_workspace/workspace/utils-deploy-quarkus-main-artifacts-to-internal-repository/devtools/project-core-extension-codestarts/target/classes/gradle-wrapper'.
22:42:51 
22:42:51 * Try:
22:42:51 > Remove any existing files in the project directory and run the init task again.
22:42:51 > Enable the --overwrite option to allow existing files to be overwritten.
22:42:51 > Run with --stacktrace option to get the stack trace.
22:42:51 > Run with --info or --debug option to get more log output.
22:42:51 > Run with --scan to get full insights.
22:42:51 > Get more help at https://help.gradle.org/.
22:42:51 
22:42:51 BUILD FAILED in 10s
22:42:51 1 actionable task: 1 executed
22:42:51 [ERROR] Command execution failed.
22:42:51 org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1)
22:42:51     at org.apache.commons.exec.DefaultExecutor.executeInternal (DefaultExecutor.java:404)
22:42:51     at org.apache.commons.exec.DefaultExecutor.execute (DefaultExecutor.java:166)
22:42:51     at org.codehaus.mojo.exec.ExecMojo.executeCommandLine (ExecMojo.java:1000)
22:42:51     at org.codehaus.mojo.exec.ExecMojo.executeCommandLine (ExecMojo.java:947)
22:42:51     at org.codehaus.mojo.exec.ExecMojo.execute (ExecMojo.java:471)
22:42:51     at com.gradle.maven.cache.extension.d.c.a (SourceFile:27)
22:42:51     at com.gradle.maven.cache.extension.d.n.a (SourceFile:23)
22:42:51     at com.gradle.maven.cache.extension.d.h.a (SourceFile:28)
22:42:51     at com.gradle.maven.cache.extension.d.m.a (SourceFile:27)
22:42:51     at com.gradle.maven.cache.extension.d.a.c (SourceFile:118)
22:42:51     at com.gradle.maven.cache.extension.d.a.a (SourceFile:62)
22:42:51     at com.gradle.maven.cache.extension.d.e.a (SourceFile:27)
22:42:51     at com.gradle.maven.cache.extension.d.l.a (SourceFile:17)
22:42:51     at com.gradle.maven.cache.extension.d.d.a (SourceFile:47)
22:42:51     at com.gradle.maven.cache.extension.d.b.a (SourceFile:26)
22:42:51     at com.gradle.maven.cache.extension.d.f$1.run (SourceFile:35)
22:42:51     at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute (SourceFile:29)
22:42:51     at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute (SourceFile:26)
22:42:51     at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute (SourceFile:66)
22:42:51     at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute (SourceFile:59)
22:42:51     at org.gradle.internal.operations.DefaultBuildOperationRunner.execute (SourceFile:157)
22:42:51     at org.gradle.internal.operations.DefaultBuildOperationRunner.execute (SourceFile:59)
22:42:51     at org.gradle.internal.operations.DefaultBuildOperationRunner.run (SourceFile:47)
22:42:51     at com.gradle.maven.cache.extension.d.f.a (SourceFile:31)
22:42:51     at com.gradle.maven.cache.extension.d.k.a (SourceFile:80)
22:42:51     at com.gradle.maven.cache.extension.h.b.lambda$createProxy$0 (SourceFile:79)
22:42:51     at jdk.proxy38.$Proxy121.execute (Unknown Source)
22:42:51     at com.gradle.maven.scan.extension.internal.d.a.executeMojo (SourceFile:116)
22:42:51     at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:370)
22:42:51     at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
22:42:51     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
22:42:51     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
22:42:51     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
22:42:51     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
22:42:51     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
22:42:51     at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
22:42:51     at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
22:42:51     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:294)
22:42:51     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
22:42:51     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
22:42:51     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:960)
22:42:51     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
22:42:51     at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
22:42:51     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
22:42:51     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
22:42:51     at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
22:42:51     at java.lang.reflect.Method.invoke (Method.java:568)
22:42:51     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
22:42:51     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
22:42:51     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
22:42:51     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
``` 
mvn command used:
` mvn -V -B -s nexus-deploy-settings.xml deploy -Dquickly -Prelocations -Dinsecure.repositories=WARN`

I've executed same mvn commands on a previous commit (33d34131c6f60d4ce3322fd973616addf9ea18a9) with gradle 8.8 and the build succeed without problems: 
```
BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  8.356 s
[INFO] Finished at: 2024-07-24T13:16:58Z
[INFO] ------------------------------------------------------------------------
[INFO] 13 goals, 12 executed, 1 from cache
```

### Expected behavior

BUILD SUCCESS

### Actual behavior

```
22:42:40 [INFO] --- exec-maven-plugin:3.1.0:exec (generate-gradle-wrapper) @ quarkus-project-core-extension-codestarts ---
22:42:41 To honour the JVM settings for this build a single-use Daemon process will be forked. For more on this, please refer to https://docs.gradle.org/8.9/userguide/gradle_daemon.html#sec:disabling_the_daemon in the Gradle documentation.
22:42:42 Daemon will be stopped at the end of the build 
22:42:51 > Task :init FAILED
22:42:51 
22:42:51 FAILURE: Build failed with an exception.
22:42:51 
22:42:51 * What went wrong:
22:42:51 Execution failed for task ':init'.
22:42:51 > Aborting build initialization due to existing files in the project directory: '/home/hudson/hudson_workspace/workspace/utils-deploy-quarkus-main-artifacts-to-internal-repository/devtools/project-core-extension-codestarts/target/classes/gradle-wrapper'.
22:42:51 
22:42:51 * Try:
22:42:51 > Remove any existing files in the project directory and run the init task again.
22:42:51 > Enable the --overwrite option to allow existing files to be overwritten.
22:42:51 > Run with --stacktrace option to get the stack trace.
22:42:51 > Run with --info or --debug option to get more log output.
22:42:51 > Run with --scan to get full insights.
22:42:51 > Get more help at https://help.gradle.org/.
22:42:51 
22:42:51 BUILD FAILED in 10s
22:42:51 1 actionable task: 1 executed
22:42:51 [ERROR] Command execution failed.
22:42:51 org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1)
22:42:51     at org.apache.commons.exec.DefaultExecutor.executeInternal (DefaultExecutor.java:404)
22:42:51     at org.apache.commons.exec.DefaultExecutor.execute (DefaultExecutor.java:166)
22:42:51     at org.codehaus.mojo.exec.ExecMojo.executeCommandLine (ExecMojo.java:1000)
22:42:51     at org.codehaus.mojo.exec.ExecMojo.executeCommandLine (ExecMojo.java:947)
22:42:51     at org.codehaus.mojo.exec.ExecMojo.execute (ExecMojo.java:471)
22:42:51     at com.gradle.maven.cache.extension.d.c.a (SourceFile:27)
22:42:51     at com.gradle.maven.cache.extension.d.n.a (SourceFile:23)
22:42:51     at com.gradle.maven.cache.extension.d.h.a (SourceFile:28)
22:42:51     at com.gradle.maven.cache.extension.d.m.a (SourceFile:27)
22:42:51     at com.gradle.maven.cache.extension.d.a.c (SourceFile:118)
22:42:51     at com.gradle.maven.cache.extension.d.a.a (SourceFile:62)
22:42:51     at com.gradle.maven.cache.extension.d.e.a (SourceFile:27)
22:42:51     at com.gradle.maven.cache.extension.d.l.a (SourceFile:17)
22:42:51     at com.gradle.maven.cache.extension.d.d.a (SourceFile:47)
22:42:51     at com.gradle.maven.cache.extension.d.b.a (SourceFile:26)
22:42:51     at com.gradle.maven.cache.extension.d.f$1.run (SourceFile:35)
22:42:51     at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute (SourceFile:29)
22:42:51     at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute (SourceFile:26)
22:42:51     at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute (SourceFile:66)
22:42:51     at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute (SourceFile:59)
22:42:51     at org.gradle.internal.operations.DefaultBuildOperationRunner.execute (SourceFile:157)
22:42:51     at org.gradle.internal.operations.DefaultBuildOperationRunner.execute (SourceFile:59)
22:42:51     at org.gradle.internal.operations.DefaultBuildOperationRunner.run (SourceFile:47)
22:42:51     at com.gradle.maven.cache.extension.d.f.a (SourceFile:31)
22:42:51     at com.gradle.maven.cache.extension.d.k.a (SourceFile:80)
22:42:51     at com.gradle.maven.cache.extension.h.b.lambda$createProxy$0 (SourceFile:79)
22:42:51     at jdk.proxy38.$Proxy121.execute (Unknown Source)
22:42:51     at com.gradle.maven.scan.extension.internal.d.a.executeMojo (SourceFile:116)
22:42:51     at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:370)
22:42:51     at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
22:42:51     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
22:42:51     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
22:42:51     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
22:42:51     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
22:42:51     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
22:42:51     at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
22:42:51     at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
22:42:51     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:294)
22:42:51     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
22:42:51     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
22:42:51     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:960)
22:42:51     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
22:42:51     at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
22:42:51     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
22:42:51     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
22:42:51     at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
22:42:51     at java.lang.reflect.Method.invoke (Method.java:568)
22:42:51     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
22:42:51     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
22:42:51     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
22:42:51     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
``` 

### How to Reproduce?

1. Just with mvn command `mvn -V -B -f devtools/project-core-extension-codestarts -s nexus-deploy-settings.xml deploy -Dquickly -Prelocations -Dinsecure.repositories=WARN`

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/devtools/project-core-extension-codestarts/pom.xml
+++ b/devtools/project-core-extension-codestarts/pom.xml
@@ -61,6 +61,7 @@
                                 <argument>--type</argument>
                                 <argument>basic</argument>
                                 <argument>--no-daemon</argument>
+                                <argument>--overwrite</argument>
                             </arguments>
                             <workingDirectory>target/classes/gradle-wrapper</workingDirectory>
                             <addOutputToClasspath>true</addOutputToClasspath>

